File: pbmx_ffi/java/io/rmf/pbmx/Xof.java
package io.rmf.pbmx;

import io.rmf.pbmx.ffi.LibPbmx;
import io.rmf.pbmx.ffi.RawXof;
import java.nio.ByteBuffer;

public final class Xof {
    RawXof raw;

    Xof(RawXof raw) {
        this.raw = raw;
    }

    public void read(ByteBuffer buffer) {
        int r = LibPbmx.INSTANCE.pbmx_read_xof(this.raw.val(), buffer, buffer.remaining());
        assert r != 0;
    }
}

File: pbmx_droid/app/src/main/java/io/rmf/pbmx/Xof.kt
package io.rmf.pbmx

import io.rmf.pbmx.ffi.FFI
import io.rmf.pbmx.ffi.RawXof
import java.nio.ByteBuffer

class Xof internal constructor(private var raw: RawXof) {

    fun read(buffer: ByteBuffer) {
        val r = FFI.pbmx_read_xof(this.raw.value(), buffer, buffer.remaining().toLong())
        assert(r != 0)
    }
}

